src/lib/_workshop-internals/components/hint-overlay/HintOverlay.svelte

```diff
@@ -0,0 +1,76 @@
+<script lang="ts">
+	/***
+	 * NOT PART OF WORKSHOP
+	 *
+	 * It should not be necessary to familiarize oneself with this file for the workshop.
+	 */
+
+	import { say } from 'cowsay2';
+	import { onMount } from 'svelte';
+	import { hintStore } from '$lib/_workshop-internals/stores/hint-store';
+	import type { Hints } from '$lib/_workshop-internals/types/hints';
+
+	let hint: string;
+	let hints: Hints;
+
+	hintStore.subscribe((hintKey) => {
+		console.log({ hintKey, hints });
+		if (hintKey) {
+			hint = say(hints[hintKey], { nowrap: true });
+		}
+	});
+
+	onMount(async () => {
+		const hintsRaw = await fetch('/hints.js');
+		const hintsText = await hintsRaw.text();
+		hints = eval(hintsText.replace('export const ', ''));
+	});
+
+	const handleCloseHint = () => {
+		hintStore.set(undefined);
+		hint = undefined;
+	};
+</script>
+
+{#if hint}
+	<div class="hint-container" on:click={handleCloseHint}>
+		<div class="rpgui-container framed hint" on:click|stopPropagation>
```

Comment:
I think the stopProgagation i stopping the user from selecting text. Might be an issue is there is something you would want to copy. Not very important, just worth noting

Reply:
Actually, it was the rpgui framework adding `user-select: none` to all containers, but nice catch! Fixed it so it's possible to select text.
(The `stopPropagation` is to prevent clicking the box from closing the hint, but clicking outside it will close the hint.)